crates/admin/src/cluster_controller/grpc_svc_handler.rs

```diff
@@ -199,15 +199,41 @@ impl ClusterCtrlSvc for ClusterCtrlSvcHandler {
         let request = request.into_inner();
 
         let extension = match request.extension {
-            Some(ext) => Some(ChainExtension {
-                segment_index_to_seal: ext.segment_index.map(SegmentIndex::from),
-
-                provider_kind: ext
-                    .provider
-                    .parse()
-                    .map_err(|_| Status::invalid_argument("Provider type is not supported"))?,
-                params: ext.params.into(),
-            }),
+            Some(ext) => {
+                if !ext.params.is_empty() {
+                    // `params`` is no longer supported. It's better to fail loudly
+                    // then act unexpectedly on invalid request version
+                    return Err(Status::invalid_argument(
+                            "Detected a deprecated argument. Please upgrade to the latest version of the command-line tool to ensure compatibility.",
```

Comment:
Maybe better to be explicit about which tool we mean?

```suggestion
                            "Detected a deprecated argument. Please upgrade to the latest version of the restatectl tool to ensure compatibility.",
```